src/package/components/ConnectWallet.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react"
+import { useWeb3 } from "@3rdweb/hooks"
+
+import ConnectWeb3Modal from "./ConnectWeb3Modal"
+
+type ButtonProps = { text: string, onClick: () => void }
+
+type Props = {
+    autoOpen: boolean
+    showDisconnect?: boolean
+    renderButton?: (props: ButtonProps) => JSX.Element
+}
+
+/**
+ * ConnectWallet 
+ * @param param0 - autoOpen - If true, the modal will be opened on page load
+ * @param param0 - showDisconnect - If true, a button will be shown to disconnect
+ * @param param0 - renderButton - If provided, you can control how a button is rendered
+ * @returns {JSX.Element}
+ */
+export const ConnectWallet = ({ autoOpen = false, showDisconnect = false, renderButton }: Props): JSX.Element => {
+
+    const [open, setOpen] = useState(autoOpen)
+
+    const { address, disconnectWallet } = useWeb3()
+
+    return <>
+        <ConnectWeb3Modal open={open} setOpen={setOpen} />
+
+        {!address && (
+            renderButton
+                ? renderButton({ text: "Connect Wallet", onClick: () => setOpen(true) })
+                : <button onClick={() => setOpen(true)}>Connect Wallet</button>
+        )}
+
+        {showDisconnect && address && (
```

Comment:
Consider moving disconnect logic to a `<ProfileView />` component
<img width="247" alt="image" src="https://user-images.githubusercontent.com/22137836/174589606-cf6f2402-8707-42b9-a457-75f53d34206e.png">
<img width="490" alt="image" src="https://user-images.githubusercontent.com/22137836/174589261-d2867067-9604-4f20-89ee-d86743de2828.png">